File: Moderate/16-2_word_frequencies.cpp
/**
 * @file
 * @brief Find the frequency of occurences of a given word in a book.
 * 
 * This module desigans and implements a method that scans a book
 * and couputes how many times a specific word occurs.
 * 
 * bonus: What if we were running this algorithm multiple times
 */


// need example book
// need frequency finder method. How to pass the book text ?

// Single Query example

#include <string>

int getFrequency (String[] book, String word) {
    word = word.trim().toLowerCase();
    int count = 0;
    for (String w : book) {
        if (w.trim().toLowerCase().equals(word)) {
            count++;
        }
    }
    return count;
}



int main() {

    return 0;
}

File: Moderate/16-2_word_frequencies.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
@file 16-2_word_frequencies.py
@brief find the frequency of words in a book
"""

def getFrequency(book, word) -> int:
    word.strip().lower()
    count = 0
    for w in book.split():
        if w.strip().lower() == word:
            count += 1
    return count


def main() -> None: 
    book = "This is the book you've been looking for"
    word = "book"

    print(getFrequency(book, word))

if __name__ == "__main__":
    main()